lib/kafka_ex/stream.ex

```diff
@@ -0,0 +1,44 @@
+defmodule KafkaEx.Stream do
+  @moduledoc false
+  alias KafkaEx.Protocol.OffsetCommit.Request, as: OffsetCommitRequest
+  alias KafkaEx.Protocol.Fetch.Request, as: FetchRequest
+
+  defstruct worker_name: nil, fetch_request: %FetchRequest{}, consumer_group: nil
+
+  defimpl Enumerable do
+    def reduce(data, acc, fun) do
+      next_fun = fn offset ->
```

Comment:
Can we make this an actual function and/or refactor it into a composition of smaller functions?